### PR TITLE
Add CJS / UMD support to `transpileES6`

### PR DESCRIPTION
**Pull request by seanpdoyle**
https://github.com/tildeio/htmlbars/pull/330

### CODE REVIEW

File: lib/utils/transpile-es6.js [hunk 1]
```diff
@@ -7,8 +7,21 @@ var babel = require('babel-core');
...
-  var outputTree = map(tree, '**/*.js', function(content, relativePath) {
+  var outputTree = map(tree, '**/*.js', function(content, relativePath, options) {
+    if (!options) {
+      options = {};
+    }
+    var transpiler;
     var moduleName = moduleNames[relativePath];
+    var format = options.format || 'amd';
+
+    if (format.match(/cjs/i)) {
```
Comment (rwjblue) [inline lib/utils/transpile-es6.js]: Why use a regexp and not just `format === 'cjs'`?

Comment (seanpdoyle) [inline lib/utils/transpile-es6.js]: handles `"CJS"`? I don't know, is that too ambitious / forgiving?

File: lib/utils/transpile-es6.js [hunk 2]
```diff
@@ -4,8 +4,22 @@ var esperanto = require('esperanto');
...
+module.exports = function(tree, description, options) {
+  if (!options) {
+    options = {};
+  }
+
+  var transpiler;
   var moduleNames = {};
+  var format = options.format || 'amd';
+
+  if (format === 'cjs') {
```
Comment (seanpdoyle) [inline lib/utils/transpile-es6.js]: @rwjblue :ok_hand:

**Comment by seanpdoyle** [comment]
@rwjblue might want to hold off on merging until https://github.com/tildeio/htmlbars/pull/331 is green and ok'd

**Pull request merged**